### PR TITLE
(RTC 498253) Added a log message if docker image is not present in co…

### PR DESCRIPTION
**Pull request by kathashok**
…ntainer image list

_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
Added a log message if docker image is not present in container image list.

### REFERENCES
Reference URL for issue tracking (JIRA/HSD/Github): https://rtc.intel.com/ccm0006001/web/projects/SSEA-%20IOTG%20PED%20Team#action=com.ibm.team.workitem.viewWorkItem&id=498253
- [x] **_Added label to the Pull Request following the template: ISDM\_\<Complexity>\*_** \
	Note-1: Depending on complexity of code changes, use the suitable word for complexity: Low/Medium/High \
	Example: PR for Slim boot loader project with medium complexity can have the label as: ISDM_Medium	
- [x] Added label to the Pull Request for easier discoverability and search
- [x] RTC or HSD number will be included in final merge. HSD must always be included if available.
- [ ] Changelogs are updated (or N/A if not customer visible)
- [x] inbm/log_changes.txt and inbm-vision/log_changes.txt are updated for potentially Validation-breaking log changes (or N/A if none)

### CODE MAINTAINABILITY
- [x] **_Commit Message meets guidelines as indicated in the URL\*_**
	- (https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md)
- [x] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):
- [x] Run 'go fmt' or autopep8 as applicable.
- [ ] New/modified methods and functions should have type annotations on signatures as applicable
- [ ] New/modified methods must have appropriate doc strings (language dependent)

### APPLICATION SPECIFIC
- [ ] Does PR change default config files under /etc? If so, will application still work after an upgrade that leaves /etc alone, like a Mender upgrade?
- [ ] Is cloud UI changed? If so, are cloud definition files updated?

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approves the Pull Request for merge is required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the maintainers for ISDM compliance.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [x] **_Quality of code (At least one should be checked as applicable)\*_**
	- [x] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [x] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [x] **_Static code scan report shows zero critical issues\*_**
- [x] Integration tests are passing

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

### SECURITY CHECKS
Please check if your PR fulfills the following requirements:

- [ ] Follow best practices when handling primitive data types
- [ ] Configure minimal permissions when opening pipes and ports
- [ ] Check contents within input structures are valid before use
- [ ] All forms of input validated
- [ ] Avoid inter-process race conditions
- [ ] Error and exception handling implemented
- [ ] Defend against Canonical Representation Issues - Any paths utilized?
- [ ] Follow 'secure by default' - Any configuration values added
- [ ] Fail safe - Any failure scenarios?
- [ ] Clean up temporary files - Any temporary files being used?

# _Code must act as a teacher for future developers_

### CODE REVIEW

**Comment by gblewis1** [comment]
@kathashok please address review checklist item

- [ ] inbm/log_changes.txt and inbm-vision/log_changes.txt are updated for potentially Validation-breaking log changes (or N/A if none)

**Comment by gblewis1** [comment]
@kathashok also please address review checklist item

- [ ] Changelogs are updated (or N/A if not customer visible)

**Comment by kathashok** [comment]
> @kathashok also please address review checklist item
> 
> * [ ]  Changelogs are updated (or N/A if not customer visible)

**Comment by nmgaston** [comment]
@kathashok - Please don't reopen this.  Just create a new PR.

**Comment by kathashok** [comment]
> @kathashok also please address review checklist item
> 
> * [ ]  Changelogs are updated (or N/A if not customer visible)

This is new PR created for RTC 498253. Change log is already updated as a part of old PR 173. 
Reference: https://github.com/intel/intel-inb-manageability/blob/develop/inbm/Changelog.md#:~:text=RTC%20498253%20%2D%20Fix%20duplicate%20of%20DBS%20remove%20operation%20in%20docker%2Dcompose%20OTA

File: inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ def _remove_images(self, ids: Any) -> None:
         for image_id in ids:
             if image_id in self.container_image_list:
                 self._remove_single_image(image_id)
+            else:
+                self._dispatcher_callbacks.broker_core.telemetry('DBS Security issue raised on imageID: '
+                                                                 + str(image_id)
+                                                                 + '.  Image is not present in container image list.')
+
         self.container_image_list[:] = []
```
Comment (gblewis1) [inline inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py:82]: document purpose of self.container_image_list on line 36 and consider choosing a more descriptive variable.  also describe invariants: what does container_image_list contain at all times? how do methods maintain invariant/invariants? i only see a clear and an append in this file.

basically, someone reviewing an individual method or the whole file should be able to follow the logic without jumping around a lot or mentally translating variable names.

**Comment by gblewis1** [comment]
@nmgaston can you comment on this also?

File: inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py (hunk 1 above)
Comment (kathashok) [reply inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py]: Renamed container_image_list variable and added a description for the same variable.  Also renamed another variable image_name. 

 **how do methods maintain invariant/invariants? i only see a clear and an append in this file.**
-> Currently we are performing only clear and append on container_image_list.

Comment (kathashok) [reply inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py]: @gblewis1 Could you please provide comments on the latest changes.

**Review by gblewis1 — CHANGES_REQUESTED** [review]

**Comment by gblewis1** [comment]
looks like inbm-vision build is failing with this PR

File: inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py [hunk 2]
```diff
@@ -135,7 +140,7 @@ def _remove_container(self, ids: Any) -> None:
                 if not image_name in str(out) or "DBS" in container_id:
```
[the remark this answers is not in the document]
Comment (kathashok) [reply inbm/dispatcher-agent/dispatcher/remediationmanager/remediation_manager.py]: Renamed and updated the code changes in latest commit.

**Comment by kathashok** [comment]
> looks like inbm-vision build is failing with this PR

Yes. Still inbm-vision build is failing with latest commit. However it is not because of this PR.

**Comment by kathashok** [comment]
> @kathashok please address review checklist item
> 
> * [ ]  inbm/log_changes.txt and inbm-vision/log_changes.txt are updated for potentially Validation-breaking log changes (or N/A if none)

Updated inbm-vision/log_changes.txt file. We have took reference of inbm-vision/Changelog.md file to edit this log_changes.txt file.

**Review by gblewis1 — APPROVED** [review]

**Review by nmgaston — APPROVED** [review]

**Pull request merged**